src/metabase/util/honey_sql_1.clj

```diff
@@ -129,6 +129,17 @@
          :when     (some? component)]
      (u/qualified-name component))))
 
+(defn identifier->name
+  "Given an identifer return its last component.
+  (identifier->name (identifier :field :metabase :user :email))
+  => :email)
+  "
+  [identifier]
+  (-> identifier
+      :components
+      last
+      keyword))
```

Comment:
Applying `keyword` to the last component here returns a namespaced keyword if the component string contains a forward slash. And column names with forward slashes are valid. Is there a reason why you can't leave string components as strings?